netbox/netbox/api/serializers/base.py

```diff
@@ -13,7 +13,48 @@
 )
 
 
+class BaseNetBoxHyperlinkedIdentityField(serializers.HyperlinkedIdentityField):
```

Comment:
The parent class should define a `get_view_name()` method. This could raise a NotImplementedError, or it could obviate the need for a separate NetBoxURLHyperlinkedIdentityField class.